Core/Models/PointMassBase.hpp

```diff
@@ -91,7 +92,7 @@ namespace gtfo
             const Eigen::Matrix<Scalar, 2, Dimensions> state = (Eigen::Matrix<Scalar, 2, Dimensions>() << DynamicsModelBase::position_.transpose(), DynamicsModelBase::velocity_.transpose()).finished();
 
             // Step the dynamics to determine our next state
-            const Eigen::Matrix<Scalar, 2, Dimensions> new_state = A_discrete_ * state + B_discrete_ * force_input.transpose();
+            const Eigen::Matrix<Scalar, 2, Dimensions> new_state = A_discrete_ * state + B_discrete_ * force_input.transpose() + C_discrete_ * parameters_.virtual_initial_position;
```

Comment:
Inside `PointMassBase`, the equations of motion have no physical connection, so this should just be `+ C_discrete_`. Furthermore, since `Parameters` is a template argument, we cannot be sure that `virtual_initial_position` will be a field. The only thing that is known inside `PointMassBase` is that `Parameters` contains a `dt` field.